pkg/api/v1/atlasdeployment_types.go

```diff
@@ -328,21 +349,30 @@ func (specArgs ProcessArgs) IsEqual(newArgs interface{}) bool {
 			if specValue.IsNil() {
 				continue
 			}
+			specValue = specValue.Elem()
+		}
+
+		if newValue.Kind() == reflect.Ptr {
 			if newValue.IsNil() {
 				return false
 			}
-			specValue = specValue.Elem()
 			newValue = newValue.Elem()
 		}
 
-		if specValue.Interface() != newValue.Interface() {
+		if stringValue(specValue.Interface()) != stringValue(newValue.Interface()) {
 			return false
 		}
 	}
 
 	return true
 }
 
+func stringValue(v interface{}) string {
+	return regexp.
+		MustCompile(`\.[0]*$`).
```

Comment:
`MustCompile` can be called once, not every time you call this method

Reply:
Good point 👌